File: 1-Basics/1.2-Basic_Maths/solution_in_cpp/2-count_digits_evenly_divides.cpp
#include<bits/stdc++.h>
using namespace std;

int evenlyDivides(int n){
    int count = 0;
    int rem = n;
    while(rem!=0){
        int temp = rem%10;
        rem = rem/10;
        if(temp>0 && n%temp==0) count++;
    }
    
    return count;
}

int main(){
    cout<<"Enter Digit N : ";
    int n;
    cin>>n;

    cout<<evenlyDivides(n);

    return 0;
}

File: 1-Basics/1.2-Basic_Maths/solution_in_python/2-count_digits_evenly_divides.py
def count_digits(n):
    count = 0
    rem = n
    while(rem != 0):
        temp = rem % 10
        rem = rem//10
        if(temp>0 and n%temp==0):
            count +=1
    
    return count


n = int(input("Enter Digit N : "))

print(count_digits(n))